src/main/asciidoc/index.adoc

```diff
@@ -116,6 +116,38 @@ A mail is sent as follows:
 {@link examples.MailExamples#sendMail}
 ----
 
+== DKIM Signature Signing emails
+
+It supports http://dkim.org[DomainKeys Identified Mail (DKIM)] Signature signing to secure your emails. All you need to
+do is to specify required configurations to sign your email.
+
+A mail client with DKIM feature enabled can be created as follows:
+
+[source,$lang]
+----
+{@link examples.MailExamples#createDKIMMailClient}
+----
+
+After the mail client is created, each `mailClient.sendMail` call will have the email signed by adding additional
+`DKIM-Signature` header.
+
+=== Caching the Attachment Streams used in DKIM
+
+To be able to perform DKIM sign, it needs to go through the email to calculate the body hash, including attachments.
+If an attachment comes from a ReadStream, it won't be able to go through again. This client provides 2 strategies to
+cache the attachment data.
+
+* Caching in memory
+
+By default, it tries to cache the data from attachment stream to a buffer so that it can be used for real email sending.
```

Comment:
By default, the stream content is fully buffered in memory so it can be actually sent later.

Reply:
Done